data/scripts/dependency-resolver.py

```diff
@@ -191,16 +191,33 @@ def handle_ccode_check(args, conf, context):
         context.add_kconfig("HAVE_%s" % dep, "bool", "n")
 
 def handle_exec_check(args, conf, context):
-    dep = conf["dependency"].upper()
-    path = which(conf["exec"]) or None
+    dep = conf.get("dependency")
+    exe = conf.get("exec")
+
+    if not exe:
+        print("Could not parse dependency: %s, no exec was specified.")
+        exit(1)
+
+    path = which(exe)
+    required = conf.get("required")
+
+    if required and not path:
+        req_label = context.find_makefile_var("NOT_FOUND")
+        req_label += "executable: %s\\n" % exe
+        context.add_append_makefile_var("NOT_FOUND", req_label, True)
 
-    context.add_cond_makefile_var(dep, path)
+    context.add_cond_makefile_var(dep.upper(), path)
```

Comment:
but answering your question, yep, it may be not upper, the defines use this same function and my be lower, in this case it's up to what is set in the json file, it's set as is.